### PR TITLE
Initial dev branch

### PR DESCRIPTION
**Pull request by ericgraf**
This PR:

- adds/changes/removes etc

### Testing

Description on how {APP-NAME} can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "labels.common" -}}
+app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+app.kubernetes.io/version: {{ .Chart.Version | quote }}
+application.giantswarm.io/team: {{ index .Chart.Annotations "application.giantswarm.io/team" | quote }}
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/_helpers.tpl:24]: don't need this in dependencies chart

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    {{- include "labels.common" . | nindent 4 }}
+data:
+  vcloud-csi-config.yaml: |+
+    vcd:
+      host: {{ .Values.configmap.site }}
+      org: {{ .Values.configmap.org }}
+      vdc: {{ .Values.configmap.ovdc }}
+      vAppName: {{ .Values.configmap.vAppName }}
+    clusterid: {{ .Values.configmap.clusterid }}
+immutable: true
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/configmap.yaml:16]: Is this required?

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  vAppName: "" # "name" field of VCDCluster object in management cluster
+  clusterid: "" # "Infra Id" field of VCDCluster object in management cluster
+
+secret:
+  create: true # Set to false if you create the secret somewhere else
+  username: ""
+  password: ""
+  refreshToken: ""
+
+statefulset:
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml:14]: Don't need to add statefulset or daemonset as a top level here. k8s resources names are not used here. Something that is more meaningful / more human readable should be used.  IE.  csi-controller

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+description: A helm chart for deploying the CPI of VMware Cloud Director (VCD) infrastructure provider (CAPVCD).
+keywords:
+- capvcd
+- vmware
+- cloud
+- cpi
+- director
+- vcd
+annotations:
+  application.giantswarm.io/team: "rocket"
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/Chart.yaml:17]: Top level charts.yaml if this is to be submitted upstream.

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+maintainers:
+- name: Team Rocket
+  email: team-rocket@giantswarm.io
+restrictions:
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/Chart.yaml:21]: Not sure about this either. The app should only be installed already on the correct installations.

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+configmap:
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml:1]: more meaningful label here would be nice

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    startIP: "192.168.8.2" # Will disappear in future versions
+    endIP: "192.168.8.100" # Will disappear in future versions
+
+secret:
+  create: true # Set to false if you create the secret somewhere else
+  username: ""
+  password: ""
+  refreshToken: ""
+
+deployment:
```
Comment (ericgraf) [inline helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml:20]: more meaningful label here aswell

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/configmap.yaml (hunk 2 above)
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/configmap.yaml:16]: It's in the upstream manifests, I'll parameterize it

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/controllerplugin-statefulset.yaml [hunk 8]
```diff
@@ -0,0 +1,98 @@
+kind: StatefulSet
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/templates/controllerplugin-statefulset.yaml:1]: I wondered that myself, I might ask maintainers to see if there's a reason.

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
+configmap:
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml:1]: changed to csiConfigmap

File: helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml (hunk 3 above)
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-director-named-disk-csi-driver/values.yaml:14]: Changed to `controllerStatefulset`

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/templates/configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+      ports:
+        http: 80
+        https: 443
+      network: {{ .Values.configmap.ovdcNetwork }}
+      vipSubnet: {{ .Values.configmap.vipSubnet }}
+      certAlias: {{ .Values.configmap.clusterid }}-cert
+      enableVirtualServiceSharedIP: {{ .Values.configmap.enableVirtualServiceSharedIP }}
+    clusterid: {{ .Values.configmap.clusterid }}
+    vAppName: {{ .Values.configmap.vAppName }}
+immutable: true
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/templates/configmap.yaml:27]: parameterized

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml (hunk 6 above)
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml:1]: changed to cpiConfigmap

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml (hunk 7 above)
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/values.yaml:20]: changed to ccmDeployment

File: helm/cloud-provider-cloud-director-app/values.yaml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+secret:
+  create: true
+
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/values.yaml:3]: good point 😄

File: helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/templates/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            - /opt/vcloud/bin/cloud-provider-for-cloud-director
+            - --cloud-provider=vmware-cloud-director
+            - --cloud-config=/etc/kubernetes/vcloud/vcloud-ccm-config.yaml
+            - --allow-untagged-cloud=true
+          volumeMounts:
+            - name: vcloud-ccm-config-volume
+              mountPath: /etc/kubernetes/vcloud
+            - name: vcloud-ccm-vcloud-basic-auth-volume
+              mountPath: /etc/kubernetes/vcloud/basic-auth
+      tolerations:
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cloud-provider-cloud-director-app/charts/cloud-provider-for-cloud-director/templates/deployment.yaml:39]: done

**Review by vxav — APPROVED** [review]

**Pull request closed without merging**